mlir/include/mlir/Dialect/Affine/Utils.h

```diff
@@ -243,16 +244,18 @@ LogicalResult replaceAllMemRefUsesWith(Value oldMemRef, Value newMemRef,
                                        ArrayRef<Value> symbolOperands = {},
                                        bool allowNonDereferencingOps = false);
 
-/// Rewrites the memref defined by this alloc op to have an identity layout map
-/// and updates all its indexing uses. Returns failure if any of its uses
-/// escape (while leaving the IR in a valid state).
+/// Rewrites the memref defined by alloc or reinterpret_cast op to have an
+/// identity layout map and updates all its indexing uses. Returns failure if
+/// any of its uses escape (while leaving the IR in a valid state).
 template <typename AllocLikeOp>
 LogicalResult normalizeMemRef(AllocLikeOp *op);
 extern template LogicalResult
 normalizeMemRef<memref::AllocaOp>(memref::AllocaOp *op);
 extern template LogicalResult
 normalizeMemRef<memref::AllocOp>(memref::AllocOp *op);
```

Comment:
Doc comment is missing anything on why a pointer to the derived op type is being passed.

Reply:
It is not clear to me as well. I just followed the exisiting convention (used pointer for reinterpret_cast op as we alrady had pointers for alloc ops).

Reply:
But it is an input argument or output argument or both? Is it nullable or not? It's time to update such a key thing in the doc comment.

Reply:
I've fixed that here: https://github.com/llvm/llvm-project/pull/134466